modules/router/src/Router.ts

```diff
@@ -175,7 +184,8 @@ export default class ConduitDefaultRouter extends ManagedModule<Config> {
 
   private async recoverFromState() {
     const r = await this.grpcSdk.state!.getKey('router');
-    if (!r || r.length === 0) return;
+    const proxyRoutes = await models.RouterProxyRoute.getInstance().findMany({});
+    if (!r || r.length === 0 || !proxyRoutes || proxyRoutes.length === 0) return;
```

Comment:
This condition will stop recovering state if the proxy routes or the regular routes are empty. This is wrong, if either of them exist we should recover them.